src/ui/SimpleFilter/SimpleFilter.ts

```diff
@@ -0,0 +1,259 @@
+      this.checkboxContainer.el.appendChild(result.checkbox.getElement());
+    });
+
+    select.el.appendChild(selectText.el);
+    select.el.appendChild(this.buildCircleElement().el);
+    select.el.appendChild(icon.el);
+    this.element.appendChild(select.el);
+    this.element.appendChild(this.checkboxContainer.el);
+    this.createWrapper().append(this.element);
+
+    $$(this.element).on('click', (e: Event) => this.handleClick(e));
+    this.bind.onRootElement(BreadcrumbEvents.populateBreadcrumb, (args: IPopulateBreadcrumbEventArgs) => this.handlePopulateBreadcrumb(args));
+    // this.bind.onRootElement(BreadcrumbEvents.clearBreadcrumb, () => this.setStateEmpty());
+    this.bind.onRootElement(QueryEvents.buildingQuery, (args: IBuildingQueryEventArgs) => this.handleBuildingQuery(args));
+    this.bind.onRootElement(QueryEvents.querySuccess, (args: IQuerySuccessEventArgs) => this.groupBy(args));
+    this.bind.onRootElement(QueryEvents.doneBuildingQuery, (args: IDoneBuildingQueryEventArgs) => this.handleDoneBuildingQuery(args));
+  }
+
+  private handleBuildingQuery(args: IBuildingQueryEventArgs) {
+    Assert.exists(args);
+    Assert.exists(args.queryBuilder);
+    const selectedValues = this.getSelectedValues();
+    if (selectedValues.length > 0) {
+      args.queryBuilder.advancedExpression.addFieldExpression(this.options.field.toLocaleString(), '==', selectedValues);
+    }
+  }
+
+  public toggle() {
+    this.expanded = $$(this.checkboxContainer).hasClass('coveo-checkbox-container-expanded');
+    if (!this.expanded) {
+      this.open();
+    } else if (this.expanded) {
+      this.close();
+    }
+  }
+
+  private handleClick(e: Event) {
+    if (e.target == this.element || e.target == $$(this.element).find('.coveo-simplefilter-select') ||
+      e.target == $$(this.element).find('.coveo-circle') || e.target == $$(this.element).find('.coveo-icon')
+      || e.target == $$(this.element).find('.coveo-simplefilter-selecttext')) {
+
+      e.stopImmediatePropagation();
+      this.toggle();
+    }
+  }
+
+
+  private open() {
+    $$(this.element).addClass('coveo-checkbox-container-expanded');
+    this.checkboxContainer.addClass('coveo-checkbox-container-expanded');
+    if ($$(document.body).find('.coveo-dropdown-background') == null) {
+      let backdrop = $$('div', { className: 'coveo-dropdown-background' });
+      backdrop.on('click', () => this.close());
+      this.element.parentElement.appendChild(backdrop.el);
+    } else {
+      const dropdown = $$(document.body).find('.coveo-dropdown-background');
+      $$(dropdown).on('click', () => this.close());
+    }
+  }
+
+  private close() {
+    $$(this.element).removeClass('coveo-checkbox-container-expanded');
+    this.checkboxContainer.removeClass('coveo-checkbox-container-expanded');
+    if ($$(this.element.parentElement).find('.coveo-dropdown-background') != null && $$(document.body).find('.coveo-checkbox-container-expanded') == null) {
```

Comment:
Erhmm.. ?

On the above code in the `open` method you are searching the `dropdown-background` from the body. Here you are searching from `this.element.parentElement`. This means that any modification to the dom generated by your component, or if it's placed somewhere else in the page other than the precise location where you are testing it will break.

In any case, you should have an utilitarian method to `getDropdownBackground`. And more method to interact with the background if needed `addBackground`, `removeBackground` etc.